src/CredentialsImport.js

```diff
@@ -0,0 +1,33 @@
+/**
+ * Use this as place to store any state or functionality related to password import promotion
+ */
+class CredentialsImport {
+    /** @param {import("./DeviceInterface/InterfacePrototype").default} device */
+    constructor (device) {
+        this.device = device
+    }
+
+    /**
+     * Check if password promotion prompt should be shown
+     */
+    shouldShowCredentialsImportFlow () {
+        const {credentials, credentialsImport} = this.device.settings.availableInputTypes
+        if (credentials?.username && credentials?.password) return false
```

Comment:
Should this be `||`? If either the username or password is present, it means the user already has some data for this site.

```suggestion
        if (credentials?.username || credentials?.password) return false
```

Reply:
Sorry this specific one won't be needed, it's just currently for local testing as the API should handle it already! Removing this whole section.